### PR TITLE
Migrate react-search to preview

### PR DESCRIPTION
**Pull request by emmayjiang**
This PR migrates SearchBox to preview, as part of #26648.

### CODE REVIEW

**Review by micahgodbolt — APPROVED** [review]

**Review by smhigley — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## 🕵 FluentUIV0 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v0"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/web-components/docs/api-report.md [hunk 1]
```diff
@@ -228,7 +228,7 @@ export const allComponents: {
     fluentTooltip: (overrideDefinition?: OverrideFoundationElementDefinition<FoundationElementDefinition> | undefined) => FoundationElementRegistry<FoundationElementDefinition, Tooltip>;
     fluentTreeView: (overrideDefinition?: OverrideFoundationElementDefinition<FoundationElementDefinition> | undefined) => FoundationElementRegistry<FoundationElementDefinition, typeof TreeView>;
     fluentTreeItem: (overrideDefinition?: OverrideFoundationElementDefinition<TreeItemOptions> | undefined) => FoundationElementRegistry<TreeItemOptions, Constructable<FoundationElement>>;
-    register(container?: Container | undefined, ...rest: any[]): void;
+    register(container?: Container, ...rest: any[]): void;
```
Comment (TristanWatanabe) [inline packages/web-components/docs/api-report.md:231]: The web-components package shouldn't be affected by this change
```suggestion
    register(container?: Container | undefined, ...rest: any[]): void;
```

File: apps/public-docsite-v9/.storybook/main.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ module.exports = /** @type {Omit<import('../../../.storybook/main'), 'typescript
     '../../../packages/react-components/react-migration-v0-v9/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
     '../../../packages/react-components/react-migration-v8-v9/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
     '../../../packages/react-components/react-datepicker-compat/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
+    '../../../packages/react-components/react-search-preview/stories/**/@(index.stories.@(ts|tsx)|*.stories.mdx)',
```
Comment (sopranopillow) [inline apps/public-docsite-v9/.storybook/main.js:18]: Can you make an issue for v-build about how we should handle this? And maybe post it in their channel as well, you can cc me if you would like to.

File: scripts/storybook/src/utils.js [hunk 3]
```diff
@@ -168,6 +168,7 @@ function _createCodesandboxRule(allPackageInfo = getAllPackageInfo()) {
       '@fluentui/react-datepicker-compat',
       '@fluentui/react-migration-v8-v9',
       '@fluentui/react-migration-v0-v9',
+      '@fluentui/react-search-preview',
```
Comment (sopranopillow) [inline scripts/storybook/src/utils.js:171]: you should also mention this in the issue.

**Review by sopranopillow — APPROVED** [review]

**Pull request merged**